### PR TITLE
Ruby 2.4 fixnum deprecation

### PR DESCRIPTION
**Pull request by dillonwelch**

### CODE REVIEW

File: devise_security_extension.gemspec [hunk 1]
```diff
@@ -20,7 +20,11 @@ Gem::Specification.new do |s|
   s.require_paths = ['lib']
   s.required_ruby_version = '>= 2.1.0'
 
-  s.add_runtime_dependency 'railties', '>= 3.2.6', '< 6.0'
+  if RUBY_VERSION >= '2.4'
```
Comment (dillonwelch) [inline devise_security_extension.gemspec:23]: I wasn't sure if this was the best way to handle supporting both 2.4 and pre 2.4. Any suggestions @natebird?

Comment (natebird) [reply devise_security_extension.gemspec:23]: I think that is a good way. 👍

File: lib/devise_security_extension/models/password_archivable.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@ def validate_password_archive
 
       # validate is the password used in the past
       def password_archive_included?
-        unless deny_old_passwords.is_a? Fixnum
+        unless deny_old_passwords.is_a? 1.class
```
Comment (natebird) [inline lib/devise_security_extension/models/password_archivable.rb:19]: Did `Fixnum` change in ruby 2.4?

File: lib/devise_security_extension/models/password_expirable.rb [hunk 3]
```diff
@@ -51,6 +51,12 @@ def update_password_changed
           self.password_changed_at = Time.now if (self.new_record? or self.encrypted_password_changed?) and not self.password_changed_at_changed?
         end
 
+        def expired_password_after_numeric?
+          return @_numeric if defined?(@_numeric)
+          @_numeric ||= self.expire_password_after.is_a?(1.class) ||
+            self.expire_password_after.is_a?(Float)
+        end
```
Comment (natebird) [inline lib/devise_security_extension/models/password_expirable.rb:58]: 👍

File: lib/devise_security_extension/models/password_archivable.rb (hunk 2 above)
Comment (dillonwelch) [reply lib/devise_security_extension/models/password_archivable.rb:19]: Yup, there's just an `Integer` class now http://blog.bigbinary.com/2016/11/18/ruby-2-4-unifies-fixnum-and-bignum-into-integer.html

Comment (natebird) [reply lib/devise_security_extension/models/password_archivable.rb:19]: Interesting. I didn't catch that before.

**Pull request merged**